### PR TITLE
Update Spring Boot to 2.3.7.RELEASE

### PR DESCRIPTION
**Pull request by AmmarBikic**
The currently used Spring Boot version 2.1.4.RELEASE will only be supported until October 2020. This pull request contains all changes needed to update Spring Boot from version 2.1.4.RELEASE to 2.3.4.RELEASE.

Reflect changes from https://github.com/eclipse/hawkbit/pull/1018

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: pom.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
    <parent>
       <groupId>org.eclipse.hawkbit</groupId>
       <artifactId>hawkbit-parent</artifactId>
-      <version>0.3.0M6</version>
+      <version>0.3.0-SNAPSHOT</version>
```
Comment (schabdo) [inline pom.xml:18]: This needs to be changed to `0.3.0M7`

File: pom.xml [hunk 2]
```diff
@@ -47,7 +47,7 @@
 
    <properties>
 
-      <hawkbit.version>0.3.0M6</hawkbit.version>
+      <hawkbit.version>0.3.0-SNAPSHOT</hawkbit.version>
```
Comment (schabdo) [inline pom.xml:50]: Same here

Comment (AmmarBikic) [reply pom.xml]: Done.

File: pom.xml (hunk 1 above)
Comment (AmmarBikic) [reply pom.xml]: Done.

File: pom.xml (hunk 2 above)
Comment (AmmarBikic) [reply pom.xml]: Done.

File: pom.xml (hunk 1 above)
Comment (AmmarBikic) [reply pom.xml]: Done.

**Pull request merged**